presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -753,6 +754,26 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
         return tableNames.build();
     }
 
+    @Override
+    public List<SchemaTableName> getMaterializedViews(ConnectorSession session, SchemaTableName tableName)
+    {
+        requireNonNull(tableName, "tableName is null");
+
+        MetastoreContext metastoreContext = new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource());
+        Table table = metastore.getTable(metastoreContext, tableName.getSchemaName(), tableName.getTableName()).orElseThrow(() -> new TableNotFoundException(tableName));
+        if (!table.getTableType().equals(MANAGED_TABLE)) {
+            throw new PrestoException(NOT_SUPPORTED, "Cannot get materialized views for a non-managed table");
+        }
+        if (MetastoreUtil.isPrestoMaterializedView(table)) {
+            throw new PrestoException(NOT_SUPPORTED, "Cannot get materialized views for a materialized view");
+        }
+        ImmutableList.Builder<SchemaTableName> materializedViews = ImmutableList.builder();
+        for (String viewName : Splitter.on(",").trimResults().omitEmptyStrings().splitToList(table.getParameters().getOrDefault(SUPPORTED_MATERIALIZED_VIEWS, ""))) {
+            materializedViews.add(new SchemaTableName(tableName.getSchemaName(), viewName));
```

Comment:
We probably need to store schema and table names for the referenced MVs. Using the original table's schema is error prone.